scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/NameDetectAnalyzer.kt

```diff
@@ -0,0 +1,268 @@
+package com.getbouncer.scan.payment.analyzer
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.RectF
+import android.media.MediaScannerConnection
+import android.media.MediaScannerConnection.OnScanCompletedListener
+import android.provider.MediaStore
+import android.util.Log
+import com.getbouncer.scan.framework.Analyzer
+import com.getbouncer.scan.framework.AnalyzerFactory
+import com.getbouncer.scan.framework.ml.ssd.hardNonMaximumSuppression
+import com.getbouncer.scan.payment.ml.AlphabetDetect
+import com.getbouncer.scan.payment.ml.SSDObjectDetect
+import com.getbouncer.scan.payment.ml.SSDOcr
+import com.getbouncer.scan.payment.ml.ssd.DetectionBox
+import com.getbouncer.scan.payment.ml.toObjectDetectionCroppedImage
+import kotlinx.coroutines.CoroutineScope
+import java.util.*
+
+
+private const val NUM_PREDICTION_STRIDES = 10
+private const val NMS_THRESHOLD = 0.75F
+private const val CHAR_CONFIDENCE_THRESHOLD = 0.7
+
+
+class NameDetectAnalyzer private constructor(
+    private val ssdObjectDetect: SSDObjectDetect?,
```

Comment:
Instead of composing the `SSDObjectDetect` analyzer here, can we make this analyzer take in the result of `SSDObjectDetect` as part of its input? This way, we can more cleanly compose the two in the `PaymentCardOcrAnalyzer`.

Reply:
are you suggesting we pull the obj detector out into PaymentCardOcrAnalyzer? We're probably moving to the world where we have a generic text detector that we'll have to apply some logic to to extract the name (and expiry) from. The way I'm thinking about it is we keep all of the logic to extract expiry and name together, and the fact that we use obj detector at all is an implementation detail that will likely change

thoughts?

Reply:
That makes a lot of sense. Thinking about the organization of the coffee that way, I'm fine leaving this as-is